src/newgrf_engine.cpp

```diff
@@ -610,8 +610,13 @@ static uint32 VehicleGetVariable(Vehicle *v, const VehicleScopeResolver *object,
 		case 0x4A:
 			switch (v->type) {
 				case VEH_TRAIN: {
+					extern RailtypeInfo _railtypes[RAILTYPE_END];
+
 					RailType rt = GetTileRailType(v->tile);
-					return (HasPowerOnRail(Train::From(v)->railtype, rt) ? 0x100 : 0) | GetReverseRailTypeTranslation(rt, object->ro.grffile);
+					RailtypeInfo *rti = &_railtypes[rt];
+					return ((rti->flags & RTFB_CATENARY) ? 0x200 : 0) |
```

Comment:
tempted to suggest these are complicated enough to be split into separate variables, to make it clearer which values go where. Nesting terneries inside bit-math is a recipe for confusion later...

Reply:
I'm not quite following this argument.

Reply:
as in have variables like `catenary_flag` & `power_flag` factored out before or-ing them together `catenary_flag | power_flag | ...`
Would look clearer, imo

Reply:
I prefer the ? : in place, like it currently is in this PR, and like it is for most variables and also for p1 and p2 in commands.

Usually you read this code in reverse, from the NewGRF point of view. So you want to know where does 0x100 come from. Distributing this information over multiple expressions makes it harder to read.

Reply:
Fair enough, I wasn't looking at it in the wider context